src/consts/consts.jl

```diff
@@ -27,18 +27,23 @@ macro const_ref(name, T, expr)
     :(const $(esc(name)) = Ref{$T}())
 end
 
-@static if MPIPreferences.abi == "MPICH"
-    include("mpich.jl")
-elseif MPIPreferences.abi == "OpenMPI"
-    include("openmpi.jl")
-elseif MPIPreferences.abi == "MicrosoftMPI"
-    include("microsoftmpi.jl")
-elseif MPIPreferences.abi == "MPItrampoline"
-    include("mpitrampoline.jl")
-elseif MPIPreferences.abi == "HPE MPT"
-    include("mpt.jl")
+# If `abi_file` is empty, choose ABI file based on ABI string, otherwise load the specified file
+@static if MPIPreferences.abi_file == ""
```

Comment:
I think i would prefer if this branch was only taken if `abi == "custom"` or something similar

Reply:
I thought of the `abi_file` preference as a "one to rule them all" property that just overrides whatever is set as ABI when it comes to loading the ABI file.

I do see the merits of your suggestion, though. However, what if users set `abi_file` to something and leave `abi` unset? Or set it to something other than `custom`? Wouldn't it be confusing if the `abi_file` is then ignored? Or should this be handled as an error in the call to `use_system_binary`?

Reply:
Given that this option will be rarely used (and is untested), I would be extremely reluctant to make it the "one to rule them all".

The configuration is complicated in any case (which is part of the reason I am reluctant to add this functionality), but I would prefer that any requirements be very explicit. The logic in `use_system_binary` should be clear.